tools/http_message_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 The Gidari Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+package tools
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestNewHTTPMessage(t *testing.T) {
+	t.Parallel()
+
+	t.Run("http methods", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.Background()
+
+		for _, tcase := range []struct {
+			method string
+		}{
+			{method: http.MethodGet},
+			{method: http.MethodPost},
+			{method: http.MethodPut},
+			{method: http.MethodDelete},
+		} {
+			req, err := http.NewRequestWithContext(ctx, tcase.method, "https://someurl.com/path", nil)
```

Comment:
@axlts Can you change all references of "https:// someurl.com" to "https://foo" ? It's a little. That way if (for whatever reason) anyone tries to actually visit the referenced site they'll just get 

> This site can’t be reached

Reply:
done